pandas/tests/frame/methods/test_to_csv.py

```diff
@@ -1285,3 +1285,24 @@ def test_to_csv_na_quoting(self):
         )
         expected = '""\n""\n'
         assert result == expected
+
+    def test_to_csv_categorical_and_ea(self):
+        # GH#46812
+        df = DataFrame({"a": "x", "b": [1, pd.NA]})
+        df["b"] = df["b"].astype("Int16")
+        df["b"] = df["b"].astype("category")
+        result = df.to_csv()
+        expected_rows = [",a,b", "0,x,1", "1,x,"]
+        expected = tm.convert_rows_list_to_csv_str(expected_rows)
+        assert result == expected
+
+    def test_to_csv_categorical_and_interval(self):
+        # GH#46297
+        df = DataFrame(
+            {"a": [pd.Interval(Timestamp("2020-01-01"), Timestamp("2020-01-02"))]}
+        )
+        df["a"] = df["a"].astype("category")  # astype("object") does not raise an error
+        result = df.to_csv()
+        expected_rows = [",a", '0,"[2020-01-01, 2020-01-02]"']
```

Comment:
has the default closed argument changed on main?

on 1.3.5 and 1.4.2, the expected would be `[",a", '0,"(2020-01-01, 2020-01-02]"']` so this test would fail on backport?

```python
print(pd.__version__)
i = pd.Interval(pd.Timestamp("2020-01-01"), pd.Timestamp("2020-01-02"))
print(repr(i))
print(i)
```
```
1.4.2
Interval('2020-01-01', '2020-01-02', closed='right')
(2020-01-01, 2020-01-02]
```
```
1.5.0.dev0+958.gf7be58a477
Interval('2020-01-01', '2020-01-02', inclusive='both')
[2020-01-01, 2020-01-02]
```

Reply:
Looks like it. Inclusive was added which defaults to both, when close is not given. I could not find this in the release notes, so we should either add it or change back to right. Probably would have to deprecate first.

For this pr, I will add closed as a keyword and catch the deprecation warning for main, then we can adjust the test as a follow up

Reply:
> For this pr, I will add closed as a keyword and catch the deprecation warning for main, then we can adjust the test as a follow up

sgtm

Reply:
opened #47365

Reply:
Changed here